packages/orchestrator-ui-components/src/components/WFOSteps/WFOStepList/WFOStepList.tsx

```diff
@@ -11,6 +11,33 @@ export interface WFOStepListProps {
     startedAt: string;
 }
 
+export const stateDelta = (prev: State, curr: State) => {
+    const prevOrEmpty = prev ?? {};
+    const prevKeys = Object.keys(prevOrEmpty);
+    const currKeys = Object.keys(curr);
+    const newKeys = currKeys.filter(
+        (key) =>
+            prevKeys.indexOf(key) === -1 ||
+            JSON.stringify(prevOrEmpty[key]) !== JSON.stringify(curr[key]),
+    );
+
+    const newState = newKeys.sort().reduce((acc: State, key) => {
+        if (
+            curr[key] === Object(curr[key]) &&
+            !Array.isArray(curr[key]) &&
+            prevOrEmpty[key]
+        ) {
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            acc[key] = stateDelta(prevOrEmpty[key], curr[key]);
```

Comment:
Could you specify -- add as code comment -- why the TS ignore is needed here?

Reply:
No; it's was already in the todo. I will when it's still needed after the mentioned refactor.